sensor/common/clusterentities/store.go

```diff
@@ -89,8 +91,37 @@ func (e *Store) initMaps() {
 	e.reverseContainerIDMap = make(map[string]map[string]struct{})
 	e.publicIPRefCounts = make(map[net.IPAddress]*int)
 	e.publicIPsListeners = make(map[PublicIPsListener]struct{})
+
+	e.historicalEndpoints = make(map[string]map[net.NumericEndpoint]*entityStatus)
+	e.historicalIPs = make(map[net.IPAddress]map[string]*entityStatus)
+	e.historicalContainerIDs = make(map[string]map[ContainerMetadata]*entityStatus)
+}
+
+func (e *Store) resetMaps() {
+	e.historyMutex.Lock()
+	defer e.historyMutex.Unlock()
+	// Maps holding historical data must not be wiped on reset! Instead, all entities must be marked as historical.
+	// Must be called before the respective source maps are wiped!
+	e.resetHistoricalMapsNoLock()
+
+	e.ipMap = make(map[net.IPAddress]map[string]struct{})
+	e.endpointMap = make(map[net.NumericEndpoint]map[string]map[EndpointTargetInfo]struct{})
+	e.containerIDMap = make(map[string]ContainerMetadata)
+	e.reverseIPMap = make(map[string]map[net.IPAddress]struct{})
+	e.reverseEndpointMap = make(map[string]map[net.NumericEndpoint]struct{})
+	e.reverseContainerIDMap = make(map[string]map[string]struct{})
+	e.publicIPRefCounts = make(map[net.IPAddress]*int)
+	e.publicIPsListeners = make(map[PublicIPsListener]struct{})
+}
+
+// resetHistoricalMapsNoLock ensures that all current data is moved to history. The history will be wiped after
+// configured number of ticks
+func (e *Store) resetHistoricalMapsNoLock() {
+	// FIXME: This is probably wrong - on transition to offline, we may lose those histories
```

Comment:
Yeah, I agree, that looks like a bug. I don't think we should be reseting these maps because in this case:

1. The pod/deployment is deleted while we are offline and before the network manager ticks.
2. The endpoint in `endpointMap` is deleted and gets added to the `historicalEndpoints` (similar thing to for ips).
3. Sensor goes online before the network manager ticks.
4. The `historical` maps get reset here and we lose the historical data.
5. The manager ticks but cannot find the endpoint/ip.

☝️ The data that was not processed in a tick while sensor was offline will be lost.